src/core/MOM_barotropic.F90

```diff
@@ -4981,9 +5089,78 @@ subroutine barotropic_init(u, v, h, eta, Time, G, GV, US, param_file, diag, CS,
           CS%lin_drag_v(i,J) = wave_drag_scale * 0.5 * (lin_drag_h(i,j) + lin_drag_h(i,j+1))
         enddo ; enddo
         deallocate(lin_drag_h)
-      endif ! len_trim(wave_drag_u) > 0 .and. len_trim(wave_drag_v) > 0
-    endif ! len_trim(wave_drag_file) > 0
-  endif ! CS%linear_wave_drag
+      endif ! (len_trim(wave_drag_u) > 0 .and. len_trim(wave_drag_v) > 0)
+    endif ! (len_trim(wave_drag_file) > 0)
+  endif ! (CS%linear_wave_drag)
+
+  if (CS%linear_freq_drag) then
+    ALLOC_(CS%lin_drag_um2(IsdB:IedB,jsd:jed)) ; CS%lin_drag_um2(:,:) = 0.0
+    ALLOC_(CS%lin_drag_vm2(isd:ied,JsdB:JedB)) ; CS%lin_drag_vm2(:,:) = 0.0
+    ALLOC_(CS%lin_drag_uk1(IsdB:IedB,jsd:jed)) ; CS%lin_drag_uk1(:,:) = 0.0
+    ALLOC_(CS%lin_drag_vk1(isd:ied,JsdB:JedB)) ; CS%lin_drag_vk1(:,:) = 0.0
+
+    if (len_trim(wave_drag_file) > 0) then
+      if (.not. CS%linear_wave_drag) then
+        inputdir = "." ;  call get_param(param_file, mdl, "INPUTDIR", inputdir)
+        wave_drag_file = trim(slasher(inputdir))//trim(wave_drag_file)
+        call log_param(param_file, mdl, "INPUTDIR/BT_WAVE_DRAG_FILE", wave_drag_file)
+      endif
+
+      if (CS%use_filter_m2 .and. m2_drag_scale > 0.0) then
+        if (len_trim(m2_drag_u) > 0 .and. len_trim(m2_drag_v) > 0) then
+          call MOM_read_data(wave_drag_file, m2_drag_u, CS%lin_drag_um2, G%Domain, &
+                             position=EAST_FACE, scale=GV%m_to_H*US%T_to_s)
+          call pass_var(CS%lin_drag_um2, G%Domain)
+          CS%lin_drag_um2(:,:) = m2_drag_scale * CS%lin_drag_um2(:,:)
```

Comment:
I think that it would be clearer and more efficient if you were to include `m2_drag_scale` in the `scale=` argument of the MOM_read_data call two lines earlier, so that it became `scale=m2_drag_scale*GV%m_to_H*US%T_to_s`.

Reply:
I followed what has been done with `lin_drag_h` and `wave_drag_scale` if `CS%linear_wave_drag` is true. I made changes for both `CS%linear_wave_drag` and `CS%linear_freq_drag`.